src/metabase/util/malli/schema.clj

```diff
@@ -368,3 +368,10 @@
   (mu/with-api-error-message
    [:re u/uuid-regex]
    (deferred-tru "value must be a valid UUID.")))
+
+(defn CollectionOf
+  "Helper for creating schemas to check whether something is an instance of a collection."
+  [item-schema]
+  [:fn
+   {:error/message (format "Collection of %s" item-schema)}
+   #(and (coll? %) (every? (partial mc/validate item-schema) %))])
```

Comment:
This is actually dead code, but I think it's generally useful so I'm inclined to keep it around.

At the moment we have no way to type something that could be a set or a vector for example.

Reply:
Surprised it's not a built-in!